tasks/eth/base-001-MCP-L1/SignFromJson.s.sol

```diff
@@ -0,0 +1,420 @@
+
+    // Known EOAs to exclude from safety checks.
+    address constant l2OutputOracleProposer = 0x642229f238fb9dE03374Be34B0eD8D9De80752c5; // cast call $L2_OUTPUT_ORACLE_PROXY "PROPOSER()(address)" # https://github.com/ethereum-optimism/superchain-registry/blob/645bb0a309970f3cc03ef6ff84670fc35917772a/superchain/extra/addresses/mainnet/base.json#L6
+    address constant batchSenderAddress = 0x5050F69a9786F081509234F1a7F4684b5E5b76C9; // https://github.com/ethereum-optimism/superchain-registry/blob/72861f4bbf15482eedfbf26dffb8507cbd1d720d/superchain/extra/genesis-system-configs/mainnet/base.json#L2
+    address constant p2pSequencerAddress = 0xAf6E19BE0F9cE7f8afd49a1824851023A8249e8a; // cast call $SYSTEM_CONFIG_PROXY "unsafeBlockSigner()(address)" # https://github.com/ethereum-optimism/superchain-registry/blob/645bb0a309970f3cc03ef6ff84670fc35917772a/superchain/extra/addresses/mainnet/base.json#L10
+    address constant batchInboxAddress = 0xFf00000000000000000000000000000000008453; // https://github.com/ethereum-optimism/superchain-registry/blob/72861f4bbf15482eedfbf26dffb8507cbd1d720d/superchain/configs/mainnet/base.yaml#L11
+
+    // Hardcoded data that should not change after execution.
+    uint256 l2GenesisBlockGasLimit = 9.75e7; // https://github.com/base-org/contract-deployments/tree/main/mainnet/2024-06-17-increase-gas-limit
+    uint256 xdmSenderSlotNumber = 204; // Verify against https://github.com/ethereum-optimism/optimism/blob/e2307008d8bc3f125f97814243cc72e8b47c117e/packages/contracts-bedrock/snapshots/storageLayout/L1CrossDomainMessenger.json#L93-L99
+
+    // Data that should not change after execution, fetching during `setUp`.
+    uint256 gasPriceOracleOverhead;
+    uint256 gasPriceOracleScalar;
+    address superchainConfigGuardian;
+    uint256 l2BlockTime;
+    uint256 l2OutputOracleSubmissionInterval;
+    uint256 finalizationPeriodSeconds;
+    uint256 l2OutputOracleStartingTimestamp;
+    uint256 l2OutputOracleStartingBlockNumber;
+    address protocolVersionsOwner;
+    uint256 requiredProtocolVersion;
+    uint256 recommendedProtocolVersion;
+
+    // Other data we use.
+    uint256 systemConfigStartBlock = 17482144; // This was an input when generating input.json
+    AddressManager addressManager = AddressManager(0x8EfB6B5c4767B09Dc9AA6Af4eAA89F749522BaE2); // https://github.com/ethereum-optimism/superchain-registry/blob/4c005f16ee1b100afc08a35a2e418d849bea044a/superchain/extra/addresses/mainnet/mode.json#L9
+    Types.ContractSet proxies;
+
+    // This gives the initial fork, so we can use it to switch back after fetching data.
+    uint256 initialFork;
+
+    /// @notice Sets up the contract
+    function setUp() public {
+        proxies = _getContractSet();
+
+        // Fetch variables that are not expected to change from an older block.
+        initialFork = vm.activeFork();
+        vm.createSelectFork(vm.envString("ETH_RPC_URL"), 20139954); // This block is from Jun-21-2024 11:59:35 AM +UTC
```

Comment:
I'm not sure if we want to use a hardcoded block number, as per the sepolia deploy we can do: 

https://github.com/ethereum-optimism/superchain-ops/blob/10b7011977ba239b04f18bd8499e08c616c328a5/tasks/sep/base-001-MCP-L1/SignFromJson.s.sol#L90


```suggestion
        vm.createSelectFork(vm.envString("ETH_RPC_URL"), block.number - 10);
```